executable-spec/src/Cardano/Ledger/Spec/STS/Update/Data.hs

```diff
@@ -32,6 +133,23 @@ data State
 
 -- | Ideation signals.
 data Signal
-  = Submit SIP
+  = Submit SIPCommit SIP
   | Reveal SIP
   deriving (Eq, Ord, Show)
+
+-- | A newtype string that is an instance of `HasHash`
```

Comment:
Why can't we use regular `String`s or `Text`?

Reply:
Because we need to define an instance of HasHash, since we turn all values to strings, in order to calculate the commit. But if we define an instance of HasHash for String, then we get a warning for an orphan instance.